test/tools/cmap_spec_runner.ts

```diff
@@ -0,0 +1,338 @@
+import { expect } from 'chai';
+import { EventEmitter } from 'events';
+import * as util from 'util';
+
+import { Connection, HostAddress } from '../../src';
+import { ConnectionPool } from '../../src/cmap/connection_pool';
+import { sleep } from './utils';
+
+type CmapOperation =
+  | { name: 'start' | 'waitForThread'; target: string }
+  | { name: 'wait'; ms: number }
+  | { name: 'waitForEvent'; event: string; count: number }
+  | { name: 'checkOut'; thread: string; label: string }
+  | { name: 'checkIn'; connection: string }
+  | { name: 'clear' | 'close' | 'ready' };
+
+type CmapPoolOptions = {
+  maxPoolSize?: number;
+  minPoolSize?: number;
+  maxIdleTimeMS?: number;
+  waitQueueTimeoutMS?: number;
+};
+
+type CmapEvent = {
+  type: string;
+  address?: 42;
+  connectionId?: number;
+  options?: 42 | CmapPoolOptions;
+  reason: string;
+};
+
+const knownTestKeys: Array<keyof CmapTest> = [
```

Comment:
```suggestion
const KNOWN_TEST_KEYS: Array<keyof CmapTest> = [
```

Reply:
right now this is the same as in the uri spec runner but I agree that in this file it's not consistent with the events list constant (since I didn't rename that from the original implementation) - do we have an opinion for all caps vs regular camelcase for general test constants? @nbbeeken @durran

Reply:
Would test constants be different than regular constants?  Our convention with constants seems to be all caps and snake case

Reply:
I'd say all caps is how I would generally define constants.

Reply:
Using all caps is a good hint that something is meant to be a set of static of well known values. If we go with caps I think we should call it `TEST_KEYS`, but I don't feel strongly. If it's not part of an API then it's changeable at will 🤷

Reply:
split the difference to `CMAP_TEST_KEYS`